src/testrel.jl

```diff
@@ -113,7 +113,7 @@ end
 Expected problems are defined by a code and an optional starting line number
 Dict(:code => "name" [, :line => <number>])
 """
-const Problem = Dict{Symbol, Any}
+const Problem = Union{Dict{Symbol, Any}, Pair{Symbol, Any}}
```

Comment:
I think it would be better to keep the original form, so that one can write `Problem(...)`, or remove it and change the definition of expected_problems in test_rel/Step.

Reply:
It seems like the type here is getting in the way of the functionality and isn't performance critical, so my intuition is to leave it a bit more open. That way one could pass in `[(:code => :BLAH)]`, `[Dict(:code => :BLAH]`, or `[(; code=:BLAH)]` We just care that we can iterate over pairs to validate things it seems

Reply:
I'm completely fine with removing the `Problem` and having something more open as long as it's readable and suggests the input should be a pair/iterable container or pairs - which is a low bar.

Reply:
This is where I want a nice trait/contract system in Julia